### PR TITLE
Small final docker image

### PR DESCRIPTION
**Pull request by ekilmer**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -47,10 +47,10 @@ jobs:
     - uses: actions/checkout@v2
     - name: Build LLVM ${{ matrix.llvm }} on ${{ matrix.ubuntu }}
       run: |
-        docker build . -t docker.pkg.github.com/lifting-bits/remill/llvm${{ matrix.llvm }}-ubuntu${{ matrix.ubuntu }}-amd64:latest -f Dockerfile --build-arg DISTRO_BASE=ubuntu${{ matrix.ubuntu }} --build-arg ARCH=amd64 --build-arg LLVM_VERSION=${{ matrix.llvm }}
+        docker build . -t docker.pkg.github.com/lifting-bits/remill/llvm${{ matrix.llvm }}-ubuntu${{ matrix.ubuntu }}-amd64:latest -f Dockerfile --build-arg UBUNTU_VERSION=${{ matrix.ubuntu }} --build-arg ARCH=amd64 --build-arg LLVM_VERSION=${{ matrix.llvm }}
     - name: Test Remill with LLVM ${{ matrix.llvm }} on ${{ matrix.ubuntu }}
       run: |
-        docker run --rm docker.pkg.github.com/lifting-bits/remill/llvm${{ matrix.llvm }}-ubuntu${{ matrix.ubuntu }}-amd64:latest -c "cd build; cmake --build . --target test_dependencies && env CTEST_OUTPUT_ON_FAILURE=1 cmake --build . --target test"
+        docker run --rm docker.pkg.github.com/lifting-bits/remill/llvm${{ matrix.llvm }}-ubuntu${{ matrix.ubuntu }}-amd64:latest --arch amd64 --ir_out /dev/stdout --bytes c704ba01000000
```
Comment (artemdinaburg) [inline .github/workflows/ci.yml:53]: We should still keep the old test command as well, it builds and runs remill's semantics tests.

Comment (ekilmer) [reply .github/workflows/ci.yml:53]: This runs inside the dockerfile now. I can split it out though if you'd like

Comment (artemdinaburg) [reply .github/workflows/ci.yml:53]: Oh thats not a problem, i just missed it in the Dockerfile

File: Dockerfile [hunk 2]
```diff
@@ -1,25 +1,36 @@
...
+FROM ${BUILD_BASE} as base
+
+RUN apt-get update && \
+    apt-get install -qqy --no-install-recommends libtinfo5 libz3-4 && \
+    rm -rf /var/lib/apt/lists/*
+
+
+# Build-time dependencies go here
+FROM trailofbits/cxx-common:llvm${LLVM_VERSION}-${DISTRO_BASE}-${ARCH} as deps
 ARG LLVM_VERSION
```
Comment (artemdinaburg) [inline Dockerfile:18]: This looks unused in the actual build steps and can probably be removed, same for the `ARG LIBRARIES` below it

File: Dockerfile [hunk 3]
```diff
@@ -1,25 +1,36 @@
...
     if [ "$(uname -m)" = "x86_64" ]; then apt-get install -qqy gcc-multilib g++-multilib; fi && \
-    apt-get install -qqy ninja-build git python2.7 curl coreutils build-essential libtinfo-dev lsb-release && \
+    apt-get install -qqy ninja-build ccache git python3 curl coreutils build-essential libtinfo-dev lsb-release && \
     rm -rf /var/lib/apt/lists/*
 
-RUN mkdir -p /remill
+
+# Source code build
+FROM deps as build
+ARG LLVM_VERSION
```
Comment (artemdinaburg) [inline Dockerfile:31]: Looks unused in build steps?

**Comment by artemdinaburg** [comment]
Added comments for some small fixes. This looks like it will be a big help.

File: Dockerfile (hunk 2 above)
Comment (ekilmer) [reply Dockerfile:18]: Yes, nice catch. Was just doing some sanity checks before I pushed 🙂

**Pull request merged**